tests/install/steps/test_expansion.py

```diff
@@ -0,0 +1,61 @@
+import pathlib
+import string
+
+from pytest_bdd import scenario, then, parsers, when
+import testinfra
+import kubernetes as k8s
+import yaml
+
+# Scenarios
+@scenario('../features/expansion.feature', 'Add one node to the cluster')
+def test_cluster_expansion(host):
+    pass
+
+# When {{{
+
+@when(parsers.parse('we declare a new node on host "{hostname}"'))
+def declare_node(host, request, kubeconfig, hostname):
+    """Declare the given node in Kubernetes."""
+    ssh_config = request.config.getoption('--ssh-config')
+    node_ip = get_node_ip(hostname, ssh_config)
+    node_manifest = get_node_manifest(node_ip)
+    k8s.config.load_kube_config(config_file=kubeconfig)
+    k8s_client = k8s.client.CoreV1Api()
+    k8s_client.create_node(body=node_from_manifest(node_manifest))
+
+# }}}
+# Then {{{
+
+@then(parsers.parse('node "{hostname}" is registered in Kubernetes'))
+def check_node_is_registered(host, request, kubeconfig, hostname):
+    """Check if the given node is registered in Kubernetes."""
+    selector = 'metadata.name=={}'.format(hostname)
+    k8s.config.load_kube_config(config_file=kubeconfig)
+    k8s_client = k8s.client.CoreV1Api()
+    nodes = k8s_client.list_node(field_selector=selector).items
+    assert len(nodes) <= 1, 'too many nodes with hostname `{}`'.format(hostname)
+    assert len(nodes) == 1, '`{}` not registered in Kubernetes'.format(hostname)
+
+
+# }}}
+# Helpers {{{
+
+def get_node_ip(hostname, ssh_config):
```

Comment:
Because there is no easy way (that we can find at least) to do this basic task? Or do you have another reason in mind?

Reply:
We want to avoid running a dumb `ip address show eth1 | awk ...` on the node when we already have that information in the SSH config file. Of course, you could also add a mention that `testinfra` does not provide a clear interface to read from this SSH config file, which is why we used this strange approach. That's what I wanted to see in a comment/docstring.